lib/Frontend/ArgsToFrontendOutputsConverter.cpp

```diff
@@ -458,3 +461,75 @@ void SupplementaryOutputPathsComputer::deriveModulePathParameters(
   mainOutputIfUsable =
       canUseMainOutputForModule && !OutputFiles.empty() ? OutputFiles[0] : "";
 }
+
+static SupplementaryOutputPaths
+createFromTypeToPathMap(const TypeToPathMap *map) {
+  SupplementaryOutputPaths paths;
+  if (!map)
+    return paths;
+  const std::pair<types::ID, std::string &> typesAndStrings[] = {
+      {types::TY_ObjCHeader, paths.ObjCHeaderOutputPath},
+      {types::TY_SwiftModuleFile, paths.ModuleOutputPath},
+      {types::TY_SwiftModuleDocFile, paths.ModuleDocOutputPath},
+      {types::TY_Dependencies, paths.DependenciesFilePath},
+      {types::TY_SwiftDeps, paths.ReferenceDependenciesFilePath},
+      {types::TY_SerializedDiagnostics, paths.SerializedDiagnosticsPath},
+      {types::TY_ModuleTrace, paths.LoadedModuleTracePath},
+      {types::TY_TBD, paths.TBDPath}};
+  for (const std::pair<types::ID, std::string &> *typeAndString =
```

Comment:
You can still use for-each loops on variables statically typed as arrays! You just can't use them on pointers.

Reply:
Cool! Done.